File: examples/comparison/nested/nested.cpp
#include <cstdint>
#include <iostream>

int main() {
    std::int64_t sum = 0;
    const int n = 4000;
    const std::int64_t mod = 1000000007LL;
    for (int i = 0; i < n; ++i) {
        for (int j = 0; j < n; ++j) {
            sum = (sum + static_cast<std::int64_t>(i) * j) % mod;
        }
    }
    std::cout << sum << '\n';
    return 0;
}

File: examples/comparison/nested/nested.go
package main

import "fmt"

func main() {
	var sum int64 = 0
	const n = 4000
	const mod int64 = 1_000_000_007
	for i := 0; i < n; i++ {
		for j := 0; j < n; j++ {
			sum = (sum + int64(i*j)) % mod
		}
	}
	fmt.Println(sum)
}
